### PR TITLE
Constructor updates

### PR DESCRIPTION
**Pull request by patrickarlt**
* Implements `url` in `options` https://github.com/Esri/esri-leaflet/issues/578
* Don't document `new Thing()` for object constructors https://github.com/Leaflet/Leaflet/issues/3408#issuecomment-100196640 encourage factories `thing()`
* Rename `L.esri.Services.FeatureLayer` to `L.esri.Services.FeatureLayerService`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
 # Changelog
 
-## 0.9.0
+## 1.0.0
 
-This represents the stable release of Esri Leaflet that will be compatible with Leaflet 0.7.3. All future 0.9.X releases will be compatible with Leaflet 0.7.3 and contain only bug fixes.
```
Comment (jgravois) [inline CHANGELOG.md]: need to get rid of the 0.9.x reference within the text.

Comment (patrickarlt) [inline CHANGELOG.md]: red = removed. It all reads 1.0.0 now. But I need to update package.json

Comment (jgravois) [inline CHANGELOG.md]: wow.  please ignore me.

**Comment by jgravois** [comment]
this looks really good!

what do you think about `L.esri.Services.FeatureService` instead of `L.esri.Services.FeatureLayerService`?  the second is a bit of a mouthful.

**Comment by jgravois** [comment]
so stoked you figured out the issue with the timeouts in the DynamicMapLayer test!  i noticed our current string ( 'http://placehold.it/500x500&text=Image1' ) returns an image with no height.

looks like we should use 'http://placehold.it/500**x500**&text=Image1' instead.

**Comment by patrickarlt** [comment]
I worry that `FeatureService ` could be confused with the actual [Feature Service](http://resources.arcgis.com/en/help/arcgis-rest-api/#/Feature_Service/02r3000000z2000000/) endpoint.

**Comment by jgravois** [comment]
oh well.  i think `FeatureServiceLayer` might have made more sense, but its no big thing.

**Comment by ngoldman** [comment]
agree with @jgravois's last comment if I'm understanding things correctly. It's a layer for feature services, not a service for feature layers, right?

**Pull request merged**